### DISCUSSION TITLE
Connect-PNPOnline -URL $url -Credentials $cred

### DISCUSSION THREAD

**Discussion by ivanjrt** [pnp/powershell#184] [Q&A]
backend:
$password is created as a MFA password = https:;//aka.ms/mfa

If I as utilizing the module **_SharePointPnPPowerShellOnline_**
`Install-Module SharePointPnPPowerShellOnline`

I have no issues running the auth:
`Connect-PNPOnline -URL $url -Credentials $cred`

If I as utilizing the module "**PnP.PowerShell"** (keep in mind I uninstalled the other module)
`Install-Module -Name "PnP.PowerShell"`
`Connect-PNPOnline -URL $url -Credentials $cred`

then it gives me an error: "Credential authentication failed AADSTS......"

any  thoughts on what would it be the error?

**Comment by jansenbe** [comment]
Did you check this : https://pnp.github.io/powershell/articles/authentication.html. You need an app to make authentication work.

**Reply by ivanjrt** [reply]
Thank you jansebe, you mean like a MFA App done at https://aka.ms/mfasetup or like a registration app?

**Reply by erwinvanhunen** [reply]
You can register your own app indeed (see `Register-PnPAzureADApp`), but you can also use the PnP Provided PnP Management Shell app. Execute

```powershell
Register-PnPManagementShellAccess
```

It will step you through the steps to register the PnP provided multitenant app registration.

**Reply by ivanjrt** [reply]
will doing this will connect to Azure and create that token from its end as an **application app**?
